### PR TITLE
Question part 2 financement

### PR DESCRIPTION
**Pull request by berenice89**

### CODE REVIEW

**Review by YannYves — APPROVED** [review]
OK but way too long

File: src/App.jsx [hunk 1]
```diff
@@ -43,6 +84,129 @@ function App() {
           <Route exact path="/question3a">
             <Question3a />
           </Route>
+          <Route exact path="/question19a">
+            <Question19a />
+          </Route>
```
Comment (LoicBrassart) [inline src/App.jsx:89]: There shouldn't have been so many routes.
At the very least, you could have a single `/question/:num` with `:num` being the question's id

File: src/App.jsx [hunk 2]
```diff
@@ -8,6 +8,47 @@ import Question1 from './components/form/Question1';
...
+import Question23a from './components/form/question/part-2/23a';
+import Question23b from './components/form/question/part-2/23b';
+import Question24 from './components/form/question/part-2/24';
+import Question25 from './components/form/question/part-2/25';
+import Question26 from './components/form/question/part-2/26';
+import Question27 from './components/form/question/part-2/27';
+import Question27a from './components/form/question/part-2/27a';
+import Question27b1 from './components/form/question/part-2/27b1';
+import Question27b2 from './components/form/question/part-2/27b2';
+import Question27b3 from './components/form/question/part-2/27b3';
```
Comment (LoicBrassart) [inline src/App.jsx:32]: Does `App` really load all these components ? I thought it was `<Form>`'s job ?

File: src/components/form/Form.jsx [hunk 3]
```diff
@@ -4,12 +4,94 @@ import Question1 from './Question1';
...
+    question19a: '',
+    question19b: '',
+    question20a: '',
+    question20b: '',
+    question20c: '',
+    question20d: '',
+    question21: '',
+    question21a: '',
+    question22: '',
+    question22a: '',
```
Comment (LoicBrassart) [inline src/components/form/Form.jsx:63]: No need for this `useState` if you use Redux and it's `useSelector`

File: src/components/form/Form.jsx [hunk 4]
```diff
@@ -98,6 +180,539 @@ const Form = () => {
...
+            nextStep={nextStep}
+            prevStep={prevStep}
+            handleChangeTest={handleChangeTest}
+            values={values}
+            step={step}
+            handleSubmit={handleSubmit}
+          />
+        </div>
+      );
+    case 31:
```
Comment (LoicBrassart) [inline src/components/form/Form.jsx:196]: Shouldnn't this whole `switch(step)` be replaced by `<Form>`'s conditional rendering ?

**Review by LoicBrassart — CHANGES_REQUESTED** [review]
See comments in the review itself

**Pull request closed without merging**